### PR TITLE
Move Texar-TF under `texar/tf/`

### PR DESCRIPTION
**Pull request by huzecong**
This PR includes the following changes:

- Move entire Texar-TF codebase under `texar/tf/`
- Add `DeprecationWarning` when user imports from `texar` but not `texar.tf` in Python 3

### CODE REVIEW

File: texar/__init__.py [hunk 1]
```diff
@@ -84,28 +42,32 @@
...
-            tf.logging.set_verbosity(tf.logging.ERROR)
-        else:
-            tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
-        from texar.version import VERSION
+        import warnings
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("always", DeprecationWarning)
+            warnings.warn("Importing from `texar` is deprecated. Please import "
+                          "from `texar.tf` instead.", DeprecationWarning,
```
Comment (ZhitingHu) [inline texar/__init__.py]: Probably add to the warning msg the recommended use? sth like:

```
recommended use: import texar.tf as tx
```

File: texar/tf/__init__.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from __future__ import division
+from __future__ import print_function
+
+# pylint: disable=wildcard-import
+
+import tensorflow as tf
+
+VERSION_WARNING = (1, 13, 2)
+
+if tuple(int(x) for x in tf.__version__.split(".")) <= VERSION_WARNING:
```
Comment (TomNong) [inline texar/tf/__init__.py]: Here could be a small issue if `tf.__version__` is `1.14.0rc1`.

**Comment by ZhitingHu** [comment]
We can probably discard the PR https://github.com/asyml/texar/pull/194 due to conflicts

@TomNong can add necesary edit to this PR to include https://github.com/asyml/texar/pull/194

**Comment by huzecong** [comment]
@TomNong There's a method under `setuptools` called `setuptools.version.pkg_resources.parse_version` (or something like this). It is essentially the same as `packaging.version.parse`, and since we (and `pip`, so basically everybody) use `setuptools` anyway, we can use it instead of writing a custom function to support complex version strings.

**Comment by TomNong** [comment]
@huzecong Thanks! I'm rebasing this PR to master.

**Comment by ZhitingHu** [comment]
This PR is ready to merge once conflicts are resolved.

**Comment by TomNong** [comment]
Hi, I solved the conflicts in PR #197, based on this PR.

**Comment by huzecong** [comment]
Closing this PR since we're moving on to #197.

**Pull request closed without merging**